### PR TITLE
Lab 5 - Stringer

### LINKED ISSUE #317: Lab 5 - Stringer
Create an executable go program to make the IPAddr type implement fmt.Stringer to print the address as a dotted quad.

Eg:
`fmt.Println(IPAddr{127, 0, 0, 1})` should print `127.0.0.1`

### PR DESCRIPTION
**Pull request by kasunfdo**
Fixes #317

Review of colleague's PR #220

#### Changes proposed in this PR:
* Implement stringer for IPAddr

### CODE REVIEW

File: 05_stringer/kasunfdo/main_test.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+
+	expected := strconv.Quote("127.0.0.1\n")
+	actual := strconv.Quote(buf.String())
```
Comment (alfredxiao) [inline 05_stringer/kasunfdo/main_test.go]: Don't understand why use `strconv.Quote` here

File: 05_stringer/kasunfdo/main_test.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+			input:    IPAddr{1, 2},
+			expected: "1.2.0.0",
+		},
+		"Non-routable IP": {
+			input:    IPAddr{0, 0, 0, 0},
+			expected: "0.0.0.0",
+		},
+		"Broadcast IP": {
+			input:    IPAddr{255, 255, 255, 255},
+			expected: "255.255.255.255",
```
Comment (alfredxiao) [inline 05_stringer/kasunfdo/main_test.go:41]: Nice test cases!

**Comment by Marchitect** [comment]
I just want to say good work!

File: 05_stringer/kasunfdo/main_test.go (hunk 1 above)
Comment (nicko-lee) [reply 05_stringer/kasunfdo/main_test.go]: Although this still works but yeah you could have less code. Yeah other than this it looks good.

**Review by nicko-lee — APPROVED** [review]

File: 05_stringer/kasunfdo/main_test.go (hunk 1 above)
Comment (kasunfdo) [reply 05_stringer/kasunfdo/main_test.go]: Changed. Thank you!

**Review by nicko-lee — APPROVED** [review]

File: 05_stringer/kasunfdo/main_test.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+	assert.Equalf(t, expected, actual, "Unexpected output")
+}
+
+func TestIPAddrStringer(t *testing.T) {
+	tests := map[string]struct {
+		input    IPAddr
+		expected string
+	}{
+		"Uninitialized IP": {
```
Comment (camscale) [inline 05_stringer/kasunfdo/main_test.go:27]: Just a minor comment - I would call this "Zero value IP". Go does not have "uninitialized" values. They are all initialised to the type's zero value if not explicitly set to something else. No need to change this for this PR, just something to keep in mind.

**Review by camscale — APPROVED** [review]
LGTM. Please squash merge this so you merge only a single commit to master without your fixup commits, or rebase to squash and then force push. You'll need reapproval to merge after a rebase though.

**Pull request merged**